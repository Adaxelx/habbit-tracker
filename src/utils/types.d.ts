export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: boolean;
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}
