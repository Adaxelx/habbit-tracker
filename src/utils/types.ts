export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
  handleDayChange: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: [
    {
      day: number;
      month: number;
      year: number;
      _id: string;
    },
  ];
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
  userId: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}

export type Color = { color?: string; id?: string };

export type NavProps = {
  header: string;
  moveDate: Function;
};

export interface DayCardProps {
  day: string;
  token: string | undefined;
}
