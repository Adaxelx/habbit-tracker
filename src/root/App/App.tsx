import React from 'react';
import { HashRouter } from 'react-router-dom';
import { ThemeProvider, UserProvider, AlertProvider, RefreshProvider } from 'context';
import { Navigation, Logo, AlertsContainer } from 'components';
import GlobalStyle from 'styles/GlobalStyle';
import { Router } from '..';

const App = () => (
  <ThemeProvider>
    <RefreshProvider>
      <AlertProvider>
        <UserProvider>
          <GlobalStyle />
          <HashRouter basename={process.env.PUBLIC_URL}>
            <Logo />
            <AlertsContainer />
            <Navigation />
            <Router />
          </HashRouter>
        </UserProvider>
      </AlertProvider>
    </RefreshProvider>
  </ThemeProvider>
);

export default App;
