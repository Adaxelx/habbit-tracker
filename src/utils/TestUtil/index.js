export { default } from './TestUtil.ts';
