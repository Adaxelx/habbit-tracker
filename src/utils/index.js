export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
