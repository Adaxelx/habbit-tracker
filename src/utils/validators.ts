export const createRequiredObject = (name: string) => ({
  required: `Field ${name} is required.`,
});

export const validatePattern = (pattern: RegExp) => ({
  pattern: {
    value: pattern,
    message: 'Incorrect characters.',
  },
});

export const emailValidation = validatePattern(
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/,
);

export const createRestrictedLengthObject = (name: string) => ({
  ...createRequiredObject(name),
  minLength: {
    value: 6,
    message: `Field ${name} must have at least 6 characters.`,
  },
  maxLength: {
    value: 30,
    message: `Field ${name} must have maximum 30 characters.`,
  },
});

const timeValidFuntion = (timeLess: string | undefined, time: string | undefined) => {
  const message = 'Your time is not in boundaries.';
  if (timeLess && time) {
    const [hours, minutes] = time.split(':');
    const [hoursMin, minutesMin] = timeLess.split(':');
    if (hoursMin < hours) {
      return true;
    }
    if (hoursMin === hours) {
      if (minutesMin < minutes) {
        return true;
      }
      return message;
    }
    return message;
  }
  return message;
};

export const createTimeValidation = (min: string | undefined, max: string | undefined) => ({
  validate: {
    min: (value: string) => timeValidFuntion(min, value),
    max: (value: string) => timeValidFuntion(value, max),
  },
});
