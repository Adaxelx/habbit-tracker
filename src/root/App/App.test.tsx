import React from 'react';
import 'jest-styled-components';
import { TestUtil } from 'utils';
import App from './App';

describe('App', () => {
  let util: TestUtil;
  beforeEach(() => {
    util = new TestUtil(<App />);
  });

  it('should match snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
});
