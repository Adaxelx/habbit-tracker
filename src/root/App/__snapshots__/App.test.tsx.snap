// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  top: 1rem;
  width: 100%;
  text-align: center;
  height: 50px;
  line-height: 50px;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {
  .c0 {
    font-size: 0.75rem;
  }
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="#/"
    >
      <h1>
        Habbit tracker
      </h1>
    </a>
  </header>
  .c0 {
  position: fixed;
  height: 50px;
  width: 50px;
  top: 1rem;
  background-color: transparent;
  border: none;
  right: 1rem;
  padding: 0;
  -webkit-transition: 300ms;
  transition: 300ms;
  z-index: 1000;
}

.c0:focus {
  outline: none;
}

.c1 {
  width: 20px;
  height: 20px;
  border-radius: 50%;
  position: absolute;
  opacity: 1;
  -webkit-transition: 300ms;
  transition: 300ms;
}

.c2 {
  background-color: #B7FF96;
  top: 0;
  left: 0;
}

.c3 {
  background-color: #89FFED;
  top: 0;
  right: 0;
}

.c4 {
  background-color: #FDF194;
  bottom: 0;
  left: 0;
}

.c5 {
  background-color: #FFADFC;
  bottom: 0;
  right: 0;
}

.c6 {
  position: absolute;
  height: 3px;
  width: 50px;
  background-color: #FF9191;
  opacity: 0;
  -webkit-transition: 300ms;
  transition: 300ms;
}

.c7 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  top: 50%;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {

}

<button
    class="c0"
    data-testid="navButton"
  >
    <div
      class="c1 c2"
    />
    <div
      class="c1 c3"
    />
    <div
      class="c1 c4"
    />
    <div
      class="c1 c5"
    />
    <div
      class="c6"
    />
    <div
      class="c6 c7"
    />
  </button>
  .c0 {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0;
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  width: 100%;
  height: 100vh;
  top: 0;
  position: fixed;
  background-color: #A4FF9D;
  z-index: 500;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 40%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c2 {
  color: #A5A5A5;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  margin-bottom: 1.25rem;
  -webkit-transform: scale(2) translateY(-50px);
  -ms-transform: scale(2) translateY(-50px);
  transform: scale(2) translateY(-50px);
  opacity: 0;
  -webkit-transition: 0;
  transition: 0;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {

}

@media (max-width:320px) {

}

<nav
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="#/"
      >
        Home page
      </a>
      <a
        class="c2"
        href="#/calendar"
      >
        Your tracker
      </a>
      <a
        class="c2"
        href="#/login"
      >
        Login
      </a>
      <a
        class="c2"
        href="#/register"
      >
        Register
      </a>
    </div>
  </nav>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  height: 100vh;
  width: 100%;
  padding: 0 1rem;
}

.c4 {
  font-size: 3rem;
  margin: 1rem 0;
}

.c5 {
  color: #808080;
  text-align: justify;
  font-size: 1.5rem;
  margin-bottom: 1.75rem;
}

.c7 {
  font-size: 1.25rem;
  margin: 0.75rem;
}

.c6 {
  color: #fff;
  background-color: #FFC136;
  border: 1px solid #707070;
  padding: 1.25rem;
  font-size: 1.5rem;
  min-width: 150px;
  display: inline-block;
  text-align: center;
}

.c0 {
  min-height: 100vh;
  width: 100%;
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 1000px;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.75rem;
  }
}

@media (max-width:320px) {
  .c4 {
    font-size: 2rem;
  }
}

@media (max-width:320px) {
  .c5 {
    font-size: 1.25rem;
  }
}

@media (max-width:320px) {

}

<main
    class="c0"
  >
    <section
      class="c1 c2"
    >
      <section
        class="c1 c3"
      >
        <h2
          class="c4"
        >
          Habbit tracker
        </h2>
        <p
          class="c5"
        >
          Track your progress in very simple and personalized app
        </p>
      </section>
      <section
        class="c1"
      >
        <a
          class="c6"
          href="#/register"
        >
          Create account for free
        </a>
        <p
          class="c7"
        >
          or
        </p>
        <a
          class="c6"
          href="#/login"
        >
          Log in
        </a>
      </section>
    </section>
  </main>
</DocumentFragment>
`;
